projects/pastanaga/src/lib/calendar/date-input.component.ts

```diff
@@ -0,0 +1,70 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, EventEmitter, Input, Output } from '@angular/core';
+import { markForCheck } from '../common/utils';
+import { TextfieldCommon } from '../textfield/textfield.common';
+import { format, isValid, isToday, isYesterday, getMonth } from 'date-fns';
+
+@Component({
+    selector: 'pa-date-input',
+    templateUrl: './date-input.component.html',
+    styleUrls: ['../popup/_popup.scss', './date-input.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DateInputComponent extends TextfieldCommon {
+
+    dateInput = '';
+    isValidDate = true;
+    currentDate: Date = new Date();
+    focused = false;
+
+    @Input() dateHelp: string = '';
+    @Input() datePlaceholder: string = 'mm/dd/yyyy';
+    @Input() errorMessage: string = 'Invalid date (mm/dd/yyyy)';
```

Comment:
you also need to support same interface as calendar for the following inputs: 
- min 
- selection
- noFuture

(Support for rangeStart, rangeEnd, can be postponed)